modules/gcloud/bigquery_test.go

```diff
@@ -81,8 +87,74 @@ func ExampleRunBigQueryContainer() {
 		}
 	}
 
-	fmt.Println(val)
-
+	fmt.Println(val[0])
 	// Output:
-	// [30]
+	// 30
+}
+
+func TestBigQueryWithDataYamlFile(t *testing.T) {
+	ctx := context.Background()
+
+	testDataPath, err := filepath.Abs(filepath.Join(".", "testdata"))
+	require.NoError(t, err)
+
+	r, err := os.Open(filepath.Join(testDataPath, "data.yaml"))
+	require.NoError(t, err)
+
+	bigQueryContainer, err := gcloud.RunBigQuery(
+		ctx,
+		"ghcr.io/goccy/bigquery-emulator:0.6.1",
+		gcloud.WithProjectID("test"),
+		gcloud.WithDataYAML(r),
+	)
+	testcontainers.CleanupContainer(t, bigQueryContainer)
+	require.NoError(t, err)
+
+	projectID := bigQueryContainer.Settings.ProjectID
+
+	opts := []option.ClientOption{
+		option.WithEndpoint(bigQueryContainer.URI),
+		option.WithGRPCDialOption(grpc.WithTransportCredentials(insecure.NewCredentials())),
+		option.WithoutAuthentication(),
+		internaloption.SkipDialSettingsValidation(),
+	}
+
+	client, err := bigquery.NewClient(ctx, projectID, opts...)
+	require.NoError(t, err)
+	defer client.Close()
+
+	selectQuery := client.Query("SELECT * FROM dataset1.table_a where name = @name")
+	selectQuery.QueryConfig.Parameters = []bigquery.QueryParameter{
+		{Name: "name", Value: "bob"},
+	}
+	it, err := selectQuery.Read(ctx)
+	require.NoError(t, err)
+
+	var val []bigquery.Value
+	for {
+		err := it.Next(&val)
+		if errors.Is(err, iterator.Done) {
+			break
+		}
+		require.NoError(t, err)
+	}
+
+	require.Equal(t, int64(30), val[0])
+}
+
+//go:embed testdata/data.yaml
+var dataYaml []byte
+
+func TestBigQueryWithDataYamlFile_multiple(t *testing.T) {
+	ctx := context.Background()
+
+	bigQueryContainer, err := gcloud.RunBigQuery(
+		ctx,
+		"ghcr.io/goccy/bigquery-emulator:0.6.1",
+		gcloud.WithProjectID("test"),
+		gcloud.WithDataYAML(strings.NewReader(string(dataYaml))),
+		gcloud.WithDataYAML(strings.NewReader(string(dataYaml))), // last file will cause an error
```

Comment:
suggestion: use bytes.NewReader to avoid conversion
```suggestion
		gcloud.WithDataYAML(bytes.NewReader(dataYaml)),
		gcloud.WithDataYAML(bytes.NewReader(dataYaml)), // second use will cause an error
```